app/src/main/java/com/babylon/wallet/android/presentation/account/createaccount/CreateAccountViewModel.kt

```diff
@@ -73,55 +74,54 @@ class CreateAccountViewModel @Inject constructor(
         savedStateHandle[CREATE_ACCOUNT_BUTTON_ENABLED] = accountName.trim().isNotEmpty() && accountName.count() <= ACCOUNT_NAME_MAX_LENGTH
     }
 
-    fun onAccountCreateClick(isWithLedger: Boolean) {
+    fun onAccountCreateClick(isWithLedger: Boolean, biometricAuthProvider: suspend () -> Boolean) {
         viewModelScope.launch {
-            if (!getProfileStateUseCase.isInitialized()) {
-                generateProfileUseCase()
-                // Since we choose to create a new profile, this is the time
-                // we discard the data copied from the cloud backup, since they represent
-                // a previous instance.
-                discardTemporaryRestoredFileForBackupUseCase(BackupType.Cloud)
-            }
-
-            val onNetworkId = if (args.networkId != -1) {
-                NetworkId.from(args.networkId)
+            val isBiometricsProvided = if (!getProfileStateUseCase.isInitialized()) {
+                if (biometricAuthProvider()) {
+                    val newMnemonic = mnemonicRepository()
```

Comment:
This is not fully safe. In secure folder it can crash the app since auth provider will return true but mnemonic creation will fail. I think best would be to add explicit TODO here so I can revisit this part after sargon PR will be merged, then I will rebase "secure folder" branch and fix this scenario

Reply:
Thanks Jakub for pointing that out. I will guard against crashing, add a comment and mention that in your PR.